### PR TITLE
[tempo] Load environment variables from Secrets

### PR DESCRIPTION
**Pull request by nlamirault**
- Like in Tempo Distributed chart , Loki chart, ...:
- `FIX` Indentation for environment variables

```yaml
# -- Environment variables to add
  extraEnv:
    - name: HELLO_WORLD
      value: "Hello from the environment"
  # -- Environment variables from secrets or configmaps to add to the ingester pods
  extraEnvFrom:
    - secretRef:
      name: tempo-credentials
```

manifest:

```yaml
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: tempo
      app.kubernetes.io/instance: release-name
  serviceName: release-name-tempo-headless
  template:
    metadata:
      labels:
        app.kubernetes.io/name: tempo
        app.kubernetes.io/instance: release-name
      annotations:
        checksum/config: 02d447f70a028decddfe18b55ce2e1f8cc609d577dbc54baee369853f163f733
    spec:
      serviceAccountName: release-name-tempo
      automountServiceAccountToken: true
      containers:
      - args:
        - -config.file=/conf/tempo.yaml
        - -mem-ballast-size-mbs=1024
        image: grafana/tempo:1.5.0
        imagePullPolicy: IfNotPresent
        name: tempo
        ports:
        - containerPort: 3100
          name: prom-metrics
        - containerPort: 6831
          name: jaeger-thrift-c
          protocol: UDP
        - containerPort: 6832
          name: jaeger-thrift-b
          protocol: UDP
        - containerPort: 14268
          name: jaeger-thrift-h
        - containerPort: 14250
          name: jaeger-grpc
        - containerPort: 9411
          name: zipkin
        - containerPort: 55680
          name: otlp-legacy
        - containerPort: 4317
          name: otlp-grpc
        - containerPort: 55681
          name: otlp-http
        - containerPort: 55678
          name: opencensus
        resources:
          {}
        env:
          - name: HELLO_WORLD
            value: Hello from the environment
        envFrom:
          - name: tempo-credentials
            secretRef: null
```

### CODE REVIEW

**Review by BitProcessor — CHANGES_REQUESTED** [review]
Hi @nlamirault, thanks for your contribution 🚀 
Please bump the chart version and update the helm-docs.

File: charts/tempo/templates/statefulset.yaml [hunk 1]
```diff
@@ -76,6 +76,10 @@ spec:
           {{- if .Values.tempo.extraEnv }}
             {{- toYaml .Values.tempo.extraEnv | nindent 12 }}
           {{- end }}
+        {{- with .Values.tempo.extraEnvFrom }}
+        envFrom:
+          {{- toYaml . | nindent 12 }}
+        {{- end }}
```
Comment (zanhsieh) [inline charts/tempo/templates/statefulset.yaml:82]:
```suggestion
        {{- with .Values.tempo.extraEnvFrom }}
        envFrom:
          {{- toYaml . | nindent 10 }}
        {{- end }}
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

File: charts/tempo/templates/statefulset.yaml [hunk 2]
```diff
@@ -76,6 +76,10 @@ spec:
           {{- if .Values.tempo.extraEnv }}
             {{- toYaml .Values.tempo.extraEnv | nindent 12 }}
           {{- end }}
```
Comment (zanhsieh) [inline charts/tempo/templates/statefulset.yaml:78]:
```suggestion
          {{- with .Values.tempo.extraEnv }}
            {{- toYaml . | nindent 12 }}
          {{- end }}
```

**Review by zanhsieh — CHANGES_REQUESTED** [review]

**Review by BitProcessor — DISMISSED** [review]
Please update the helm docs

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**